graph/src/components/store/traits.rs

```diff
@@ -471,14 +471,16 @@ pub trait ChainStore: Send + Sync + 'static {
     ) -> Result<Vec<serde_json::Value>, Error>;
 
     /// Get the `offset`th ancestor of `block_hash`, where offset=0 means the block matching
-    /// `block_hash` and offset=1 means its parent. Returns None if unable to complete due to
-    /// missing blocks in the chain store.
+    /// `block_hash` and offset=1 means its parent. If `root` is passed, short-circuit upon finding
+    /// a child of `root`. Returns None if unable to complete due to missing blocks in the chain
+    /// store.
```

Comment:
The comments here should be clearer, we need short circuiting on root because  FEVM can skip blocks and relying on offset can be inaccurate, so we use root as a reference and whenever we find a block whose parent is the root, we know that this is the ancestor we are looking for. The usual value of  root is `subgraph_ptr`